### PR TITLE
Feature/dense dispatching meta

### PR DESCRIPTION
**Pull request by shubham808**
A simple cookbook that trains LDA with float64 and float32

### CODE REVIEW

File: doc/cookbook/source/examples/base_api/dense_dispatching.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
+==================================================
+Dense Dispatching for Linear Discriminant Analysis
```
Comment (karlnapf) [inline doc/cookbook/source/examples/base_api/dense_dispatching.rst]: call this "Dense feature types", remove the LDA

File: doc/cookbook/source/examples/base_api/dense_dispatching.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+==================================================
+
+Some algorithms like Linear Discriminant Analysis, Least Angle Regression, etc. in shogun support training with features of multiple primitive type.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/base_api/dense_dispatching.rst]: Be more general.
"Dense features, like those read from a sgclass(CCSVFIle), can be read into memory in multiple primitive types, e.g. 32 bit or 64 bit floating point numbers.
Downstream algorithms can often deal with multiple of such feature types, and will carry out required computations in the provided primitive type.
Reducing from 64 bit float to 32 bit float can be beneficial if for example very large matrices have to be stored.

File: doc/cookbook/source/examples/base_api/dense_dispatching.rst [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+-------
+Example
+-------
+
+Imagine we have files with training data. We create 64 bit and 32 bit float CDenseFeatures of appropriate primitive type and also create :sgclass:`CBinaryLabels` as
```
Comment (karlnapf) [inline doc/cookbook/source/examples/base_api/dense_dispatching.rst]: missing sglcass missing newline after period

File: doc/cookbook/source/examples/base_api/dense_dispatching.rst [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+.. sgexample:: dense_dispatching.sg:create_features
+
+We create an instance of :sgclass:`CLDA` and provide labels.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/base_api/dense_dispatching.rst]: We create an instance of a shogun algorithm (here sgclass LDA)

File: doc/cookbook/source/examples/base_api/dense_dispatching.rst [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+.. sgexample:: dense_dispatching.sg:create_instance
+
+We train with 64 bit RealFeatures as
```
Comment (karlnapf) [inline doc/cookbook/source/examples/base_api/dense_dispatching.rst]: Group all of the feature calls together. And say "We can pass any of the feature type to the algorithm for training"

File: src/shogun/util/factory.h [hunk 6]
```diff
@@ -65,7 +65,7 @@ namespace shogun
 		return features;
 	}
 
-	CFeatures* features(CFile* file, EPrimitiveType primitive_type = PT_FLOAT64)
+	CFeatures* features(CFile* file, machine_int_t primitive_type = PT_FLOAT64)
```
Comment (shubham808) [inline src/shogun/util/factory.h]: this is more flexibile and we also need it to pass primitives from swig

Comment (karlnapf) [reply src/shogun/util/factory.h]: I am not sure I understand why you want that change? This effectively bypasses any type checking that swig code could do at runtime (passed a valid enum type)

File: doc/cookbook/source/examples/base_api/dense_dispatching.rst [hunk 7]
```diff
@@ -0,0 +1,26 @@
+==============================
+Dense feature type Dispatching
```
Comment (karlnapf) [inline doc/cookbook/source/examples/base_api/dense_dispatching.rst]: dispatching not Dispatching

File: doc/cookbook/source/examples/base_api/dense_dispatching.rst [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+==============================
+
+Dense features, like those read from a :sgclass:`CCSVFIle`, can be read into memory in multiple primitive types, e.g. 32 bit or 64 bit floating point numbers.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/base_api/dense_dispatching.rst]: sgclass CDenseFeatures

File: doc/cookbook/source/examples/base_api/dense_dispatching.rst [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+Downstream all algorithms that inherit from :sgclass:`CDenseRealDispatch` can deal with multiples of such types, and they will carry out required computations in the corresponding primitive type.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/base_api/dense_dispatching.rst]: Algorithms that inherit from ... downstream can deal with all of such types

File: doc/cookbook/source/examples/base_api/dense_dispatching.rst (hunk 7 above)
Comment (karlnapf) [reply doc/cookbook/source/examples/base_api/dense_dispatching.rst]: this is still unaddressed

File: examples/meta/src/base_api/dense_dispatching.sg [hunk 10]
```diff
@@ -0,0 +1,21 @@
+File f_feats_double = csv_file("../../data/classifier_binary_2d_linear_features_train.dat")
+File f_feats_float = csv_file("../../data/classifier_binary_2d_linear_features_train.dat")
+File f_labels_train = csv_file("../../data/classifier_binary_2d_linear_labels_train.dat")
+
+#![create_features]
+Features features_float = features(f_feats_float, enum EPrimitiveType.PT_FLOAT64)
```
Comment (karlnapf) [inline examples/meta/src/base_api/dense_dispatching.sg]: i think double and float is compused here
double is 64 bit

File: src/shogun/util/factory.h (hunk 6 above)
Comment (karlnapf) [reply src/shogun/util/factory.h]: I understand now

File: doc/cookbook/source/examples/base_api/dense_dispatching.rst [hunk 11]
```diff
@@ -0,0 +1,26 @@
+==============================
+Dense feature type dispatching
+==============================
+
+:sgclass:`DenseFeatures`, like those read from a :sgclass:`CCSVFIle`, can be read into memory in multiple primitive types, e.g. 32 bit or 64 bit floating point numbers.
+All algorithms that inherit from :sgclass:`CDenseRealDispatch` downstream can deal with multiples of such types, and they will carry out required computations in the corresponding primitive type.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/base_api/dense_dispatching.rst]: multiple

**Comment by karlnapf** [comment]
Thanks!

**Pull request merged**